dash/dash-renderer/src/actions/callbacks.ts

```diff
@@ -340,33 +342,108 @@ function updateComponent(component_id: any, props: any) {
     };
 }
 
-function sideUpdate(outputs: any, dispatch: any) {
-    toPairs(outputs).forEach(([id, value]) => {
-        let componentId = id,
-            propName;
+function parsePMCId(id: string) {
```

Comment:
There's a lot of magic numbers in this function (`+2`, `+1`) -- could you add a comment explaining some of the string manipulations here?

Reply:
It's just the indexes of the strings, I think it's pretty readable if you know the format:
`{"key":"value"}.propName` or just the dictionary, so index refers to position of `}`, if the `index + 2  < id.length` means there is more data after the index and `.`, the propname starts at index + 2 and componentId is 0 to index + 1.

Reply:
got it, that makes sense, thanks! Could you add that format as a comment in the code?

Reply:
This looks duplicative of:

https://github.com/plotly/dash/blob/09252f8d2f690480cc468b2e015f9e2417dc90ad/dash/dash-renderer/src/actions/dependencies.js#L93

Reply:
This handle case where the prop name is not attached to the key, which comes with `set_props` so it searches for `}` instead of a dot.

Reply:
OK - can we at least put these in the same location - maybe move them both to a new file so it's easier to find them and we avoid making any more of them?

My main concern with doing this in slightly different ways is accidentally introducing additional constraints on what constitutes a valid ID, and/or causing an error with certain IDs. I think your logic here is OK, as you only get to searching for `}` after you already know the ID starts with `{` which is an [invalid character](https://github.com/plotly/dash/blob/b1c6efb08d5f80e873d95f489e0c8d65335aed84/dash/_validate.py#L107) for string IDs - but that's something we need to check carefully every time we modify the ID handling code.

Reply:
Thanks for putting the new function in a new file. The existing function is still in its old location though, so at some point please let's bring that one into the new file too.